src/services/accessibility/html_id_generator.ts

```diff
@@ -44,7 +44,7 @@ export type UseGeneratedHtmlIdOptions = {
    */
   conditionalId?: string;
 };
-export const useGeneratedHtmlId = ({
+export const useDeprecatedGeneratedHtmlId = ({
```

Comment:
We can likely skip exporting this hook
```suggestion
const useDeprecatedGeneratedHtmlId = ({
```

Reply:
Oh, I'd also love it if we added a comment on **when** we can remove this deprecated component. e.g.,

```suggestion

// We can remove this deprecated hook fully once EUI no longer needs to support React 16-17
const useDeprecatedGeneratedHtmlId = ({
```